### PR TITLE
Fix scoutfs print <data_dev> hang

### PR DESCRIPTION
**Pull request by bgly**
If a user tries to print a data device exit early if
it is data device.

Signed-off-by: Bryant G. Duffy-Ly <bduffyly@versity.com>

### CODE REVIEW

**Comment by versity-github** [comment]
Build 124 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/124/ git: a909626eb0bc12521743ab82ae455bbdbb76fc09

File: utils/src/print.c [hunk 1]
```diff
@@ -1006,6 +1006,12 @@ static int print_volume(int fd)
 
 	print_super_block(super, SCOUTFS_SUPER_BLKNO);
 
+	if (!(le64_to_cpu(super->flags) && SCOUTFS_FLAG_IS_META_BDEV)) {
+	    fprintf(stderr,
```
Comment (zabbo) [inline utils/src/print.c:1010]: full tabs for indentation, not spaces, and let's set ret for an error so it's not 0 from before.. maybe -EINVAL?

**Review by zabbo — CHANGES_REQUESTED** [review]
Nice, thanks for taking this.   In addition to the minor cleanups, let's remove that "did you mean to do this?" message from print_super if we're going to be returning an error.

**Comment by versity-github** [comment]
Build 125 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/125/ git: 0954abd45c69890613f44a0939d82d013e0c9c9d

**Comment by versity-github** [comment]
Build 126 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/126/ git: 5fe030764c791f06ec24606d8c591828012950a4

File: utils/src/print.c [hunk 2]
```diff
@@ -1006,6 +1002,13 @@ static int print_volume(int fd)
 
 	print_super_block(super, SCOUTFS_SUPER_BLKNO);
 
+	if (!(le64_to_cpu(super->flags) && SCOUTFS_FLAG_IS_META_BDEV)) {
```
Comment (zabbo) [inline utils/src/print.c:1005]: Ha!   I just noticed that this should also be "&", not "&&" :)

**Review by zabbo — CHANGES_REQUESTED** [review]

File: utils/src/print.c (hunk 2 above)
Comment (bgly) [reply utils/src/print.c]: Fixed, I left it because it works either way atm since there is only 1 flag.

**Comment by versity-github** [comment]
Build 127 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/127/ git: 95f2a87864d9899bb17f037532b91824d202e4ae

**Review by zabbo — APPROVED** [review]

**Pull request merged**